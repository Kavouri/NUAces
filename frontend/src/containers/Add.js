import React from 'react';

const Add = () => <div>Add Partners or Events</div>;

export default Add;
