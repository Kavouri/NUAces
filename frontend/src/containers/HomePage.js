import { connect } from 'react-redux';
import HomePageView from '../views/HomePageView';

export const mapStateToProps = (state) => {
  return {
    partners: state.partners.partnerList,
  };
};

const HomePage = connect(mapStateToProps)(HomePageView);

export default HomePage;
