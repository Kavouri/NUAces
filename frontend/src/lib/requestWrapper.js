import rp from 'request-promise';

const request = (relativeUrl, method, data = {}) => {
  const absoluteUrl = process.env.REACT_APP_BASE_URL + relativeUrl;
  console.log(absoluteUrl);
  const options = {
    url: absoluteUrl,
    json: true,
    data,
  };
  return rp[method.toLowerCase()](options);
}

export { request };