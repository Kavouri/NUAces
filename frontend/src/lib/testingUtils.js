import _ from 'lodash';

const mockedFunctions = {};

// Mocks a function on a parent object.
const mockFunction = (parent, functionName, mockImplementation = () => {}) => {
  const mockedFunction = jest.spyOn(parent, functionName).mockImplementation(mockImplementation);
  mockedFunctions[functionName] = mockedFunction;
  return mockedFunction;
};

// Restores all mocked functions. Call this in afterEach.
const restoreAllMocks = () => {
  _.forEach(mockedFunctions, (mockedFunction) => {
    mockedFunction.mockRestore();
  });
};

// Restores a single mock by name of the function.
const restoreSingleMock = (functionName) => {
  if (mockedFunctions[functionName]) {
    mockedFunctions[functionName].mockRestore();
  }
};

export { mockFunction, restoreAllMocks, restoreSingleMock };
