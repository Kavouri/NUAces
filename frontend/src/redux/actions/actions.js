

export const ADD_PARTNER = 'ADD_PARTNER';
export const ADD_EVENT = 'ADD_EVENT';
export const REGISTER_FOR_EVENT = 'REGISTER_FOR_EVENT';
export const SUCCESSFUL_LOGIN = 'SUCCESSFUL_LOGIN';

export const addPartner = (partnerInfo) => { return { type: ADD_PARTNER, partnerInfo }; };
export const addEvent = (eventInfo) => { type: ADD_EVENT, eventInfo };
export const registerForEvent = (eventId) => { return { type: REGISTER_FOR_EVENT, eventId }; };

//Login
export const successfulLogin = (user) => { return { type: SUCCESSFUL_LOGIN, user: user }; };
