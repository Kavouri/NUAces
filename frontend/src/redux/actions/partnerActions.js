export const REQUEST_PARTNERS = 'REQUEST_PARTNERS';
export const RECEIVE_PARTNERS = 'RECEIVE_PARTNERS';

const requestPartners = () => {
  return {
    type: REQUEST_PARTNERS,
    isFetchingPartners: true,
  };
};

const receivePartners = (partners) => {
  return {
    type: RECEIVE_PARTNERS,
    isFetchingPartners: false,
    partners,
  };
};

export { receivePartners, requestPartners };
