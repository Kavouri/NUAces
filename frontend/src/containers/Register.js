import React from 'react';
import { request } from '../lib/requestWrapper';
import _ from 'lodash';

import Name from './registration/Name';
import { Birthday, MONTHS } from './registration/Birthday';
import Email from './registration/Email';
import Password from './registration/Password';
import ConfirmPassword from './registration/ConfirmPassword';
import Error from './registration/Error';

export default class Register extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      firstName: '',
      lastName: '',
      month: 'January',
      day: '',
      year: '',
      email: '',
      password: '',
      passwordConfirm: '',
      error: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  validateForm() {
    let error = '';
    _.forEachRight(this.state, (value, key) => {
      if (key !== 'error' && !this.state[key]) {
        error = `${key.split(/(?=[A-Z])/).map(word => _.upperFirst(word)).join(' ')} field is required`;
      }
    });

    if (!error) {

      if (this.state.day.concat(this.state.year).match(/[a-z]/i)) {
        error = 'DD-YYYY contains alphabetic characters.';
      }
      else if (this.state.day.length > 2 || this.state.year.length !== 4) {
        error = 'DD-YYYY has an invalid number of characters.';
      }
      else if (this.state.password !== this.state.passwordConfirm) {
        error = 'Passwords don\'t match';
      }
      else if (this.state.email.indexOf('@') < 0) {
        error = 'Email is not valid.';
      }
    }

    if (error) {
      this.setState({
        error
      });
    }
    return !error;
  }

  handleSubmit(event) {
    event.preventDefault();
    if (this.validateForm()) {
      const registrationForm = {
        name: `${this.state.firstName} ${this.state.lastName}`,
        dob: `${this.state.year}-${MONTHS[this.state.month]}-${this.state.day}`,
        email: this.state.email,
        password: this.state.password
      };

      request('/user', 'POST', registrationForm)
        .then((res) => {
            this.props.history.push('/');
          } else {
            this.setState({ error: res.body });
          }
        })
        .catch((error) => {
          this.setState({ error: error.error });
        });
    }
  }

  render() {
    return (
      <div className="registration">
      <form className="registration-form" onSubmit={this.handleSubmit}>
        <label>Name</label>
        <Name handleChange={this.handleChange} firstName={this.state.firstName} lastName={this.state.lastName}/>
        <label>Birthday</label>
        <Birthday handleChange={this.handleChange} month={this.state.month} day={this.state.day} year={this.state.year}/>
        <label>Email</label>
        <Email handleChange={this.handleChange} email={this.state.email}/>
        <label>Password</label>
        <Password handleChange={this.handleChange} password={this.state.password}/>
        <label>Confirm Password</label>
        <ConfirmPassword handleChange={this.handleChange} passwordConfirm={this.state.passwordConfirm}/>
        <button className="inner-registration register-button" size="3" type="submit">Register</button>
        {this.state.error && <Error error={this.state.error}/>}
      </form>
    </div>
    );
  }
}
