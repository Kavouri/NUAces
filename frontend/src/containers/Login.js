import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import request from '../lib/requestWrapper';
import { successfulLogin } from '../redux/actions/actions';
import Error from './registration/Error';

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    };
    this.handleEmail = this.handleEmail.bind(this);
    this.handlePassword = this.handlePassword.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleEmail(event) {
    this.setState({ email: event.target.value });
  }

  handlePassword(event) {
    this.setState({ password: event.target.value });
  }

  validateForm() {
    const { email, password } = this.state;
    let error = '';
    if (!email) {
      error = 'Please enter your email';
    } else if (this.state.email.indexOf('@') < 0) {
      error = 'Email is not valid';
    } else if (!password) {
      error = 'Please enter your password';
    }
    if (error) {
      this.setState({ error });
    }
    return !error;
  }

  handleSubmit(event) {
    event.preventDefault();
    if (this.validateForm()) {
      const loginForm = {
        email: this.state.email,
        password: this.state.password
      };

      let baseUrl = process.env.REACT_APP_BASE_URL + "/login";
      const options = {
        url: baseUrl,
        form: loginForm
      };

      console.log(request);
      request('/login', 'POST', loginForm)
        .then((res) => {
          if (res.status == 200) {
            this.props.successfulLogin(res);
            this.props.history.push('/');
          }
        })
        .catch((error) => {
          this.setState({ error: error.error });
        });
    }
  }

  render() {
      return <div className="container-fluid">
          <h1 className="text-center">Login</h1>
          <form>
              <div className="row Form-Margin">
                  <div className="col-xs-6 col-xs-offset-3">
                      <input
                          placeholder="Email"
                          name="email"
                          type="text"
                          className="form-control text-center"
                          value={this.state.email}
                          onChange={this.handleEmail}
                      />
                  </div>
              </div>
              <div className="row Form-Margin">
                  <div className="col-xs-6 col-xs-offset-3">
                      <input
                          placeholder="Password"
                          name="password"
                          type="password"
                          className="form-control text-center"
                          value={this.state.password}
                          onChange={this.handlePassword}
                      />
                  </div>
              </div>
              <div className="row Form-Margin">
                  <div className="col-xs-6 col-xs-offset-3">
                      <button className="btn btn-block btn-primary" 
                        type="submit" onClick={this.handleSubmit}>
                          Login
                      </button>
                  </div>
              </div>
              <div className = "row Form-Margin">
                  <div className="col-xs-6 col-xs-offset-3">
                      <Link to="/register">
                          <button className="btn btn-block btn-success">
                              Register
                          </button>
                      </Link>
                  </div>
              </div>
              <div className="row Form-Margin">
                  <div className="col-xs-6 col-xs-offset-3">
                      <h2> {this.state.error 
                        && <Error error={this.state.error}/>}</h2>
                  </div>
              </div>
          </form>
      </div>;
  }
}

const mapDispatchToProps = dispatch => {
  return {
    successfulLogin: (resp) => { dispatch(successfulLogin(resp)) }
  };
};

export default Login = connect(() => {return {}}, mapDispatchToProps)(Login);



