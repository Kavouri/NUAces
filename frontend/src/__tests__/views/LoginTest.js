import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { UnwrappedLogin, mapDispatchToProps } from '../../containers/Login';
import { mockFunction, restoreAllMocks } from '../../lib/testingUtils';
import { successfulLogin } from '../../redux/actions/actions';
import * as request from '../../lib/requestWrapper';

let instance;
let mockState;
let requestStub;
beforeEach(() => {
  instance = new UnwrappedLogin({ successfulLogin: jest.fn()});
  mockFunction(instance, 'setState');
  requestStub = mockFunction(request, 'request', () => {
    return Promise.resolve({ status: 200 });
  });
  mockState = { email: 'bob@husky.neu.edu', password: 'password' };
  instance.state = mockState;
});

describe('Login', () => {
    describe('handleEmail', () => {
      it('sets the state accordingly', () => {
        instance.handleEmail({ target: { value: 'carroll.t' }});
        expect(instance.setState).toHaveBeenCalled;
        expect(instance.setState).toHaveBeenCalledWith({ email: 'carroll.t'});
      });
    });
    describe('handlePassword', () => {
      it('sets the state accordingly', () => {
        instance.handlePassword({ target: { value: 'password'}});
        expect(instance.setState).toHaveBeenCalled;
        expect(instance.setState).toHaveBeenCalledWith({ password: 'password'});
      });
    });
    describe('validateForm', () => {
      it('returns true with a validated form', () => {
        expect(instance.validateForm()).toEqual(true);
        expect(instance.setState).toNotHaveBeenCalled;
      });
      it('sets error for missing email', () => {
        instance.state.email = '';
        expect(instance.validateForm()).toEqual(false);
        expect(instance.setState).toHaveBeenCalled;
        expect(instance.setState).toHaveBeenCalledWith({ error: 'Please enter your email' });
      });
      it('sets error for invalid email', () => {
        instance.state.email = 'carroll.t';
        expect(instance.validateForm()).toEqual(false);
        expect(instance.setState).toHaveBeenCalled;
        expect(instance.setState).toHaveBeenCalledWith({ error: 'Email is not valid' });
      });
      it('sets error for missing password', () => {
        instance.state.password = '';
        expect(instance.validateForm()).toEqual(false);
        expect(instance.setState).toHaveBeenCalled;
        expect(instance.setState).toHaveBeenCalledWith({ error: 'Please enter your password' });
      });
    });
    describe('handleSubmit', () => {
      beforeEach(() => {
        mockFunction(instance, 'validateForm', () => {
          return true;
        });
      });
      it('does nothing if the form is not validated', () => {
        mockFunction(instance, 'validateForm', () => {
            return false;
          });
          instance.handleSubmit({ preventDefault: jest.fn() });
          expect(requestStub).toNotHaveBeenCalled;
      });
      it('sends the login request otherwise', () => {
        instance.handleSubmit({ preventDefault: jest.fn() });
        expect(requestStub).toHaveBeenCalled;
        expect(requestStub).toHaveBeenCalledWith('/login', 'POST', { email: 'bob@husky.neu.edu', password: 'password' });
        expect(instance.props.successfulLogin).toHaveBeenCalled;
      });
    });
    describe('matchDispatchToProps', () => {
      const mockDispatch = jest.fn();
      const trigger = mapDispatchToProps(mockDispatch).successfulLogin({ name: 'Tighe' });
      expect(mockDispatch).toHaveBeenCalled;
      expect(mockDispatch).toHaveBeenCalledWith({ type: 'SUCCESSFUL_LOGIN', user: { name: 'Tighe' }});
    });
    it('matches a snapshot if no activity has occurred', () => {
        const tree = renderer.create(<MemoryRouter><UnwrappedLogin /></MemoryRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('matches a snapshot if an empty form was submitted', () => {
        const tree = renderer.create(<MemoryRouter><UnwrappedLogin /></MemoryRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
