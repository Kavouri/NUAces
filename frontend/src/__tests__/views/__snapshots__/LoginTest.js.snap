// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches a snapshot if an empty form was submitted 1`] = `
<div
  className="container-fluid"
>
  <h1
    className="text-center"
  >
    Login
  </h1>
  <form>
    <div
      className="row Form-Margin"
    >
      <div
        className="col-xs-6 col-xs-offset-3"
      >
        <input
          className="form-control text-center"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="row Form-Margin"
    >
      <div
        className="col-xs-6 col-xs-offset-3"
      >
        <input
          className="form-control text-center"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="row Form-Margin"
    >
      <div
        className="col-xs-6 col-xs-offset-3"
      >
        <button
          className="btn btn-block btn-primary"
          onClick={[Function]}
          type="submit"
        >
          Login
        </button>
      </div>
    </div>
    <div
      className="row Form-Margin"
    >
      <div
        className="col-xs-6 col-xs-offset-3"
      >
        <a
          href="/register"
          onClick={[Function]}
        >
          <button
            className="btn btn-block btn-success"
          >
            Register
          </button>
        </a>
      </div>
    </div>
    <div
      className="row Form-Margin"
    >
      <div
        className="col-xs-6 col-xs-offset-3"
      >
        <h2>
           
        </h2>
      </div>
    </div>
  </form>
</div>
`;

exports[`Login matches a snapshot if no activity has occurred 1`] = `
<div
  className="container-fluid"
>
  <h1
    className="text-center"
  >
    Login
  </h1>
  <form>
    <div
      className="row Form-Margin"
    >
      <div
        className="col-xs-6 col-xs-offset-3"
      >
        <input
          className="form-control text-center"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="row Form-Margin"
    >
      <div
        className="col-xs-6 col-xs-offset-3"
      >
        <input
          className="form-control text-center"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      className="row Form-Margin"
    >
      <div
        className="col-xs-6 col-xs-offset-3"
      >
        <button
          className="btn btn-block btn-primary"
          onClick={[Function]}
          type="submit"
        >
          Login
        </button>
      </div>
    </div>
    <div
      className="row Form-Margin"
    >
      <div
        className="col-xs-6 col-xs-offset-3"
      >
        <a
          href="/register"
          onClick={[Function]}
        >
          <button
            className="btn btn-block btn-success"
          >
            Register
          </button>
        </a>
      </div>
    </div>
    <div
      className="row Form-Margin"
    >
      <div
        className="col-xs-6 col-xs-offset-3"
      >
        <h2>
           
        </h2>
      </div>
    </div>
  </form>
</div>
`;
